// backend/routes/api/tables.js
var Table = require('backend/models/table').Table;

exports.get = function(req, res, next) {

    return Table.find( function( err, tables ) {
        if( !err ) {
            return res.send( tables );
        } else {
            return console.log( err );
        }
    });
};

exports.post = function(req, res, next) {

    table = new Table({
        title:      req.body.title,
        posTop:     req.body.posTop,
        posLeft:    req.body.posLeft,
        count:      req.body.count
    });

    table.save(function(err, table) {
      res.send( table );
    });
};

exports.put = function(req, res, next) {

    return Table.findById( req.params.id, function( err, table ) {

        table.title     = req.body.title;
        table.posTop    = req.body.posTop;
        table.posLeft   = req.body.posLeft;

        return table.save( function( err ) {
            if( !err ) {
                console.log( 'table updated' );
            } else {
                console.log( err );
            }
            return res.send( table );
        });
    });
};
exports.delete = function(req, res, next) {

    return Table.findById( req.params.id, function( err, table ) {
        return table.remove( function( err ) {
            if( !err ) {
                console.log( 'Contact removed' );
                //return response.send( '' );
            } else {
                console.log( err );
            }
        });
    });
};
